disperser/batcher/txn_manager.go

```diff
@@ -193,8 +199,11 @@ func (t *txnManager) ensureAnyTransactionEvaled(ctx context.Context, txs []*tran
 				}
 			}
 
-			if errors.Is(err, ethereum.NotFound) {
-				t.logger.Debug("Transaction not yet mined", "component", "TxnManager", "method", "ensureAnyTransactionEvaled", "txID", tx.TxID, "txHash", tx.Hash().Hex())
+			if errors.Is(err, ethereum.NotFound) || errors.Is(err, walletsdk.ErrReceiptNotYetAvailable) {
+				t.logger.Debug("Transaction not yet mined", "component", "TxnManager", "method", "ensureAnyTransactionEvaled", "txID", txID, "txHash", tx.Hash().Hex(), "err", err)
+			} else if errors.Is(err, walletsdk.ErrTransactionFailed) {
+				t.logger.Debug("Transaction failed", "component", "TxnManager", "method", "ensureAnyTransactionEvaled", "txID", txID, "txHash", tx.Hash().Hex(), "err", err)
+				delete(txnsToQuery, txID)
```

Comment:
I guess an optimization would be to immediately return if the map is empty, but doesn't seem important.